src/common/user-store.ts

```diff
@@ -71,14 +76,32 @@ export class UserStore extends BaseStore<UserStoreModel> {
    */
   @observable downloadKubectlBinaries = true;
   @observable openAtLogin = false;
+
+  /**
+   * The column IDs under each configurable table ID that have been configured
+   * to not be shown
+   */
   hiddenTableColumns = observable.map<string, ObservableToggleSet<string>>();
 
-  protected async handleOnLoad() {
-    await this.whenLoaded;
+  /**
+   * The set of file/folder paths to be synced
+   */
+  syncKubeconfigEntries = observable.set([
+    path.join(os.homedir(), ".kube"),
+    ClusterStore.storedKubeConfigFolder,
+  ]);
+
+  async load(): Promise<void> {
+    /**
+     * This has to be here before the call to `new Config` in `super.load()`
+     * as we have to make sure that file is in the expected place for that call
+     */
+    await fileNameMigration();
+    await super.load();
 
     // refresh new contexts
-    this.refreshNewContexts();
-    reaction(() => this.kubeConfigPath, this.refreshNewContexts);
+    await this.refreshNewContexts();
```

Comment:
Just a note that we probably should remove `refreshNewContexts` because that information is not used anymore.

Reply:
Will do in follow up PR